PP-607 - Product Cards docs (#355)
livechat/livechat-public-docs: site/content/product-cards/store-api.md

---
weight: 30
---

# Store API

This section describes all endpoints that you should implement in your backend service. You can use our <a href="https://productcards.livechatinc.com/#/validator/" target="_blank">Backend Validator Tool</a> to test your implementation.

## Security

```
Access-Control-Allow-Origin: https://productcards.livechatinc.com
Access-Control-Allow-Methods: GET, POST, OPTIONS
```

To avoid CORS-related conflicts, you should set proper `Access-Control-Allow-Origin` and `Access-Control-Allow-Methods` headers.

```
Product Cards client ID: 9e63f93f6782051e6a43549c132a1f7a
```

The app uses **LiveChat OAuth token** to authorize requests sent to your backend service. To confirm that the user is authorized, you should [validate given token](../authorization/#validating-the-access-token) and check if it matches Product Cards client ID **and** your license number.

## Stores

```bash
curl -X GET \
  'https://store.com/api/stores' \
  -H 'Authorization: Bearer <livechat-access-token>'
```

> The very first request (that will be sent to your backend from our app) fetches basic info about stores available for Product Cards app. It means that it's possible to have more than one store connected to Product Cards. Widget calls your backend for a list of stores as follows:

Reviewer: ```suggestion
The very first request, which will be sent to your backend from our app, fetches the basic info about stores available for the Product Cards app. It means that it's possible to have more than one store connected to Product Cards. The widget calls your backend for a list of stores as follows:
```

Reviewer: Zastanowilabym sie czy uzywac tu slowa widget, do tej pory nie bylo ono uzyte. Moze lepiej trzymac sie 'app'.

> *__Note__: Even if you want to connect only one store, you need to create this endpoint and provide us the store ID - it's required for next requests.*

Reviewer: ```suggestion
*__Note__: Even if you want to connect only one store, you need to create this endpoint and provide us the store ID - it's required for the next requests.*
```

> Example response:

Reviewer: ```suggestion

```

```json
[
  {
    "id": "1001", // internal id of a store
    "name": "MyStore"
  },
  {
    "id": "1002",
    "name": "MyStore2"
  }
]
``` 

## Categories

> Next Product Cards will fetch information about categories available in the store. The **Categories filter** is based on this endpoint data. 

Reviewer: ```suggestion
**Product Cards** will fetch information about categories available in the store. The **category filter** is based on this endpoint data. 
```

> ```bash

Reviewer: ```suggestion
> A sample request:

```bash
```

> curl -X GET \

Reviewer: ```suggestion

> A sample request:

curl -X GET \
```

  'https://store.com/api/categories?storeId=10001' \
  -H 'Authorization: Bearer <livechat-access-token>'
```

Parameters:

> - **storeId**: is an internal ID provided by `/stores` endpoint

Reviewer: ```suggestion
- **storeId**: is an internal ID provided by the `/stores` endpoint
```

> Example response:

Reviewer: ```suggestion

```

> ```json

Reviewer: ```suggestion
> A sample response:

```json
```

> ```json

Reviewer: ```suggestion
> A sample response:

```json
```

{
  "categories": [
    {
      "id": "5001", // internal id of a category
      "name": "Books"
    },
    {
      "id": "5010",
      "name": "Games"
    }
  ]
}
```

## Products

> Last but not least, the app will fetch a list of products which show up in the form of tiles. The response should be paginated and support the following params:

Reviewer: ```suggestion
Last but not least, the app will fetch a list of products which will show up in the form of tiles. The response should be paginated and should support the following params:
```

> - **currentPage**: the page that was returned with current request

Reviewer: ```suggestion
- **currentPage**: the page that was returned with a current request
```

> - **pageCount**: total amount of pages

Reviewer: ```suggestion
- **pageCount**: the total number of pages
```

> - **totalCount**: total amount of products (from all pages)

Reviewer: ```suggestion
- **totalCount**: the total number of products from all pages
```

> ```bash

Reviewer: ```suggestion

> A sample request:

```bash
```

curl -X GET \
  'https://store.com/api/products?storeId=10001&page=1&categories=5001%2C5010&term=Harry+Potter&sort=name&direction=asc' \
  -H 'Authorization: Bearer <livechat-access-token>'
```

Parameters:

> - **storeId**: is an internal ID provided by `/stores` endpoint

Reviewer: ```suggestion
- **storeId**: an internal ID provided by the `/stores` endpoint
```

- **page** *(optional):* allows to paginate products (default: `1`)

> - **categories** *(optional):* allows to filter products by a list of categories (list of IDs separated by comma and urlencoded)

Reviewer: urlencoded?

Author: https://www.w3schools.com/tags/ref_urlencode.asp (I'll add the link here)

- **term** *(optional):* allows to filter products by name (urlencoded)
- **sort** *(optional):* allows to sort by field (supported values: `id`, `name`)

> - **direction** *(optional):* allows to determine sort's direction (supported values: `asc`, `desc`)

Reviewer: ```suggestion
- **direction** *(optional):* allows to determine the sorting order (supported values: `asc`, `desc`)
```

> Example response:

Reviewer: ```suggestion

```

> ```json

Reviewer: ```suggestion
> A sample response:

```json
```
